platform/android/export/export.cpp

```diff
@@ -198,6 +198,84 @@ static const char *android_perms[] = {
 	NULL
 };
 
+//TODO: put this somewhere better (i.e. templates folder)
+const String ANDROID_MANIFEST_TEXT = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+									 "<manifest xmlns:android=\"http://schemas.android.com/apk/res/android\"\n"
+									 "    xmlns:tools=\"http://schemas.android.com/tools\"\n"
+									 "    package=\"*PACKAGE_NAME*\"\n"
+									 "    android:versionCode=\"*VERSION_CODE*\"\n"
+									 "    android:versionName=\"*VERSION_NAME*\"\n"
+									 "    android:installLocation=\"auto\" >\n"
+									 "\n"
+									 "<instrumentation android:targetPackage=\"org.godotengine.godot\"/>\n"
+									 "\n"
+									 "    <!-- Adding custom text to the manifest is fine, but do it outside the custom USER and APPLICATION BEGIN/END comments, -->\n"
+									 "    <!-- as that gets rewritten. -->\n"
+									 "\n"
+									 "    <supports-screens\n"
+									 "        android:smallScreens=\"*SMALL_SCREENS*\"\n"
+									 "        android:normalScreens=\"*NORMAL_SCREENS*\"\n"
+									 "        android:largeScreens=\"*LARGE_SCREENS*\"\n"
+									 "        android:xlargeScreens=\"*X_LARGE_SCREENS*\" />\n"
+									 "\n"
+									 "    <!-- glEsVersion is modified by the exporter, changing this value here has no effect. -->\n"
+									 "    <uses-feature\n"
+									 "        android:glEsVersion=\"*GLES_VERSION*\"\n"
+									 "        android:required=\"true\" />\n"
+									 "\n"
+									 "*FEATURES*\n"
+									 "<!-- Custom user permissions XML added by add-ons. It's recommended to add them from the export preset, though. -->\n"
+									 "<!--CHUNK_USER_PERMISSIONS_BEGIN-->\n"
+									 "<!--CHUNK_USER_PERMISSIONS_END-->\n"
+									 "*PERMISSIONS*\n"
+									 "\n"
+									 "    <!-- Any tag in this line after android:icon will be erased when doing custom builds. -->\n"
+									 "    <!-- If you want to add tags manually, do before it. -->\n"
+									 "    <!-- WARNING: This should stay on a single line until the parsing code is improved. See GH-32414. -->\n"
+									 "    <application android:label=\"@string/godot_project_name_string\" android:allowBackup=\"false\" tools:ignore=\"GoogleAppIndexingWarning\" android:icon=\"@mipmap/icon\" >\n"
+									 "\n"
+									 "        <!-- The following metadata values are replaced when Godot exports, modifying them here has no effect. -->\n"
+									 "        <!-- Do these changes in the export preset. Adding new ones is fine. -->\n"
+									 "\n"
+									 "        <!-- XR mode metadata. This is modified by the exporter based on the selected xr mode. DO NOT CHANGE the values here. -->\n"
+									 "        <meta-data\n" //
```

Comment:
So in the case that XR isn't needed, we'll have something like `android:name=\"*XR_MODE_METADATA_NAME*\"` unchanged in the final manifest?

Instead should this section look like this:
`XR_MODE_PLACEHOLDER`

And then if there's nothing to include we replace the placeholder with a blank line, but if there is something we replace it with the whole `<meta-data` element?

Reply:
Should I do the same thing with focus awareness?